### PR TITLE
refactor: convert FormFieldWrapper to FC

### LINKED ISSUE #787: Refactor FormFieldWrapper Component
# Description
This issue is related to refactoring the `FormFieldWrapper` Component.

## Acceptance Criteria
Refactor the component from a [class component to a function component](https://reactjs.org/docs/components-and-props.html#function-and-class-components).
Feel free to contribute other improvements to the component while refactoring if you come across any opportunities. 

Refer to other components to keep formatting consistent. For example, reference the [Badge](https://github.com/dcos-labs/ui-kit/blob/main/packages/badge/components/badge.tsx), [BadgeButton](https://github.com/dcos-labs/ui-kit/blob/main/packages/badge/components/badgeButton.tsx), or [Tabs](https://github.com/dcos-labs/ui-kit/blob/main/packages/tabs/components/Tabs.tsx) components.

## How to Contribute 
Please review the [contributing documentation](https://github.com/dcos-labs/ui-kit/blob/main/CONTRIBUTING.md) beforehand. 

Steps to Contribute:
1. [Fork the repository](https://github.com/dcos-labs/ui-kit/fork).
2. Clone it to your local system.
3. Create a new branch. Please adhere to branch formatting, for example `<your_initials>/refactor/<component_name>`
4. Make your changes. Make sure that tests are still passing. To update snapshot tests, run `npm run test -- -u` and include the updated tests in your commit.
5. Commit your changes.  When committing be sure to prefix with the appropriate conventional commit type. For example, `refactor: button component`. 
6. Push up your changes. In GitHub, click the compare & pull request button to begin drafting your pull request.
7. Fill out the pull request form and attach the "ready to review" label when complete.

### PR DESCRIPTION
**Pull request by davidtaing**
# Description

Refactors FormFieldWrapper to a functional component.

fixes #787

## Which issue(s) does this PR relate to?

- #787 

## Testing

No changes to the UI

## Trade-offs

N/A

## Screenshots

N/A - No Changes to the UI

## Checklist

- [X] If any new components were added, there are exported from `packages/index.ts`
- [ ] This PR is associated with a JIRA and mentions in the commit message footer ("Closes …")
- [ ] This PR contains breaking changes and states in the commit message body ("BREAKING CHANGE: …")
- [X] I have reviewed the changes and provided detail to the sections above

### CODE REVIEW

File: packages/shared/components/FormFieldWrapper.tsx [hunk 1]
```diff
@@ -92,22 +72,33 @@ class FormFieldWrapper extends React.PureComponent<FormFieldWrapperProps, {}> {
...
+    <>
+      {children({
+        getValidationErrors: getValidationErrors(errors, id),
+        getHintContent: getHintContent(hintContent),
+        describedByIds: getDescribedBy(getHintContentId(), getErrorIds()),
+        isValid: !errors || (errors && errors.length === 0)
+      })}
+    </>
+  );
+};
```
Comment (nataliepina) [inline packages/shared/components/FormFieldWrapper.tsx:102]: We've been using React.memo() for components that were previously a `PureComponent`. 

Could you wrap this export below with React.memo()?

`export default React.memo(FormFieldWrapper); ` 

You may need to update the snapshot tests afterward - `npm run test -- -u`

**Review by nataliepina — CHANGES_REQUESTED** [review]
Was able to test out these changes and the Form related components are functioning as expected. I left one comment requesting a change otherwise.

**Review by vacas5 — COMMENTED** [review]
agree with @nataliepina here, just need to memoize this one to recreate PureComponent behavior

**Comment by davidtaing** [comment]
Just made the changes. We should be good to go.

I'm self-taught and I'm finding it difficult to the first development job. So I wanted to sincerly thank you @nataliepina & @vacas5. Thank you for giving me this opportunity. Thank you for making it easy for me to get the project up and running. And thank you for your prompt replies, prompt review and making this a great experience overall.

Please let me know if I can help out anywhere else like any other issues or even other projects that need love.

❤️

**Review by mstruebing — APPROVED** [review]
Thank you @davidtaing these changes look very good from my point of view 🚀

**Comment by nataliepina** [comment]
@davidtaing Great work! Some of these component refactors can be challenging. I will be adding a few more component refactors to the issues later today. After those, I will be adding some different types of issues. Feel free to add an issue if you come across anything. We appreciate your contributions and I'm happy to hear it was easy for you to get up and running with the project.

**Review by nataliepina — APPROVED** [review]
👏🏻

**Pull request merged**